unit/adsr.go

```diff
@@ -0,0 +1,149 @@
+package unit
+
+import (
+	"math"
+
+	"buddin.us/shaden/dsp"
+)
+
+func newAdsr(name string, _ Config) (*Unit, error) {
+	io := NewIO()
+	return NewUnit(io, name, &adsr{
+		state: &adsrState{
+			lastTrigger: -1,
+		},
+		stateFunc:  adsrIdle,
+		trigger:    io.NewIn("trigger", dsp.Float64(0)),
```

Comment:
This might need to be removed since technically only the `gate` input will be used here.

Reply:
Of course, I overlooked that one haha